idb/postgres/internal/writer/writer_test.go

```diff
@@ -1564,3 +1565,169 @@ func TestWriterAddBlock0(t *testing.T) {
 		assert.Equal(t, expected, accounts)
 	}
 }
+
+// Simulate a scenario where app boxes are created, mutated and deleted in consecutive rounds.
+func TestWriterAppBoxTableInsertMutateDelete(t *testing.T) {
```

Comment:
The only test I see missing here is adding another box after the first set, i.e. inserting box 6 during the second round.